### PR TITLE
Missing enum tests

### PR DESCRIPTION
**Pull request by bramtayl**
[Uncovered](https://codecov.io/gh/JuliaLang/julia/src/master/base/Enums.jl#L68)
[Uncovered](https://codecov.io/gh/JuliaLang/julia/src/master/base/Enums.jl#L131)
[Uncovered](https://codecov.io/gh/JuliaLang/julia/src/master/base/Enums.jl#L163)

### CODE REVIEW

**Comment by bramtayl** [comment]
Sorry for the multiple commits

**Comment by bramtayl** [comment]
CI failure unrelated?

File: test/enums.jl [hunk 1]
```diff
@@ -167,3 +169,34 @@ end
 @test Int(haggis) == 4
 
 @test (Vector{Fruit}(undef, 3) .= apple) == [apple, apple, apple]
+
+# long, discongruous
+@enum Alphabet begin
+    alphabet_a = 27
```
Comment (JeffBezanson) [inline test/enums.jl]: I'm not sure this will hit the intended line, since the values are still continuous (`length(values) == hi - lo + 1`).

Comment (bramtayl) [reply test/enums.jl]: Ok that makes sense, I bumped it up to 28

Comment (JeffBezanson) [reply test/enums.jl]: That doesn't change anything; the test `length(values) == hi - lo + 1` will still be true.

Comment (bramtayl) [reply test/enums.jl]: Ok sorry. How should I change it? I was thinking that alphabet_b would be assigned to 1, so that (27 - 1 + 1) > 26

Comment (bramtayl) [reply test/enums.jl]: Ok sorry. How should I change it? I was thinking that alphabet_b would be assigned to 1, so that `(27 - 1 + 1) > 26`

Comment (JeffBezanson) [reply test/enums.jl]: Ah, I see. The default value is the previous value plus one:
```
julia> @enum X a=10 b c=20 d

julia> a
a::X = 10

julia> b
b::X = 11

julia> c
c::X = 20

julia> d
d::X = 21
```

Comment (bramtayl) [reply test/enums.jl]: Ok, I think this new version will work based on the output of `@macroexpand`

**Comment by bramtayl** [comment]
Bump?

**Comment by StefanKarpinski** [comment]
Seems good to me. (And thanks for the additional test coverage! Always great to have.) @JeffBezanson, you happy with this now?

**Comment by bramtayl** [comment]
Yay! I was thinking for my next PR I'd try to incorporate a bunch of tests at once. Would that be easier or harder to review?

**Comment by StefanKarpinski** [comment]
I would keep the set of tests focused on one thing. Too many changes make PRs harder to review. Note the variance in how long these coverage PRs take to get merged: if you combined them then the combined PR would take at least as long to get merged as the longest of the individual ones.

**Comment by bramtayl** [comment]
Ok

**Pull request merged**